src/app/calculator/calculator.component.ts

```diff
@@ -14,65 +15,160 @@ import { MOCK } from '../shared/mockdata';
 export class CalculatorComponent {
   constructor() {
     this.motocycles = MOCK.models;
-    this.canvasLeg = new p5(this.sketch, 'canvas-human');
+    this.canvasLeg = new p5(this.sketchLeg, 'canvas-human');
     this.canvasLeg.setup = () => {
-      this.canvasLeg.createCanvas(1000, 560);
+      this.canvasLeg.createCanvas(1000, 660).position(0, 0);
       this.canvasLeg.noLoop();
     };
+    this.canvasTors = new p5(this.sketchTorsAndArm, 'canvas-human');
+    this.canvasTors.setup = () => {
+      this.canvasTors.createCanvas(1000, 660).position(0, 0);
+      this.canvasTors.noLoop();
+    };
   }
 
+  canvasLeg;
+
+  canvasTors;
+
+  riderValues = {
+    height: null,
+    heightRiderPixel: null,
+    headPixel: null,
+    neckPixel: null,
+    torsPixel: null,
+    palmPixel: null,
+    armPixel: null,
+    legPixel: null,
+    waistToKneePixel: null,
+    kneeToFootPixel: null,
+    footPixel: null,
+    coordinateWaist: {
+      x: null,
+      y: null,
+    },
+    coordinateKnee: {
+      x: null,
+      y: null,
+    },
+    coordinateFootOnGround: {
+      x: null,
+      y: null,
+    },
+    coordinatePalmCenter: {
+      x: null,
+      y: null,
+    },
+    coordinateShoulder: {
+      x: null,
+      y: null,
+    },
+  };
+
+  motocycles: MotorcycleModel[];
+
   dataForm = new FormGroup({
-    motocycle: new FormControl('', Validators.required),
-    heightRider: new FormControl('', Validators.required),
-    legLength: new FormControl('', Validators.required),
+    motocycle: new FormControl('', this.validatorMotocycle),
+    heightRider: new FormControl('', this.validatorHeightRider),
   });
 
-  private canvasLeg;
+  validatorMotocycle(control: AbstractControl) {
+    if (!(control.value)) {
+      return {
+        isError: true,
+      };
+    }
+    return null;
+  }
 
-  motocycles: MotorcycleModel[];
+  validatorHeightRider(control: AbstractControl) {
+    if (!(control.value)) {
+      return {
+        isError: true,
+      };
+    }
+    if ((Number(control.value) < 100) || (Number(control.value) > 250)) {
+      return {
+        isError: true,
+      };
+    }
+    return null;
+  }
 
-  sketch = (p: any) => {
+  sketchLeg = (p: any) => {
```

Comment:
what does mean "**p**" and why it's "**any**"?

Reply:
i so the example https://stackoverflow.com/questions/49472433/how-to-use-p5-js-in-angular-5-application/53464162
it works
I think here p means p5 object (used by p5.js library for work with canvas)
I just change any to p5 and it's works)

Reply:
1) so use any - bad practice in general
2) Example - it's example, but you use real. If you understand what I mean
So keep as is check this later